apps/web/src/components/milestone-logs/MilestoneTable.tsx

```diff
@@ -41,6 +42,18 @@ export const MilestoneTable = ({ milestones }: MilestoneTableProps) => {
     );
   };
 
+  const getDuration = (milestone: ApplicantStatusAuditRO): string => {
+    const start = milestone.start_date;
+    let end = milestone.end_date;
+    if (!end) {
+      const nextMilestone = milestones.filter(m => m.status.id > milestone.status.id)[0];
```

Comment:
We should sort based on the date and get the next milestone date here.
- We are hoping that ID is in incremental order and,
- assuming that no one gets the latest date on any previous milestone.
If both of the above cases are in place this implemented logic will work fine.

Approved it as per MVP requirements.

Reply:
I noticed that `milestones` are filtered by the selected tab and am going to fix it.
MelestoneTable is just for milestones except ones under recruitment category and they should come from ATS. So condition of `assuming that no one gets the latest date on any previous milestone` shouldn't be on us. With this condition, I am going to sort milestones by `start_date.`